### ISSUE TITLE
WARNING:root:Caution: Your LLM is currently in training mode, ensuring accurate gradient computation. Please be vigilant, particularly regarding BatchNorm and Dropout operations.

### ISSUE THREAD

**Issue by dragen1860** [NVlabs/VILA#28]
hi, dear author,
   when finetunning on my custom dataset, the LLM is keeping printing:
```
WARNING:root:Caution: Your LLM is currently in training mode, ensuring accurate gradient computation. Please be vigilant, particularly regarding BatchNorm and Dropout operations.
WARNING:root:Caution: Your LLM is currently in training mode, ensuring accurate gradient computation. Please be vigilant, particularly regarding BatchNorm and Dropout operations.
WARNING:root:Caution: Your LLM is currently in training mode, ensuring accurate gradient computation. Please be vigilant, particularly regarding BatchNorm and Dropout operations.
WARNING:root:Caution: Your LLM is currently in training mode, ensuring accurate gradient computation. Please be vigilant, particularly regarding BatchNorm and Dropout operations.
WARNING:root:Caution: Your LLM is currently in training mode, ensuring accurate gradient computation. Please be vigilant, particularly regarding BatchNorm and Dropout operations.
WARNING:root:Caution: Your LLM is currently in training mode, ensuring accurate gradient computation. Please be vigilant, particularly regarding BatchNorm and Dropout operations.
```

when check the code, it says:
```python
    def freezed_module_patch(self):
        '''
        Huggingface will call model.train() at each training_step. To ensure the expected behaviors for modules like dropout, batchnorm, etc., we need to call model.eval() for the freezed modules.
        '''
        if self.training:
            if self.get_llm() and not getattr(self.config, "tune_language_model", False):
                logging.warning("Caution: Your LLM is currently in training mode, ensuring accurate gradient computation. Please be vigilant, particularly regarding BatchNorm and Dropout operations.")
            if self.get_vision_tower() and not getattr(self.config, "tune_vision_tower", False):
                self.get_vision_tower().eval()
            if self.get_mm_projector() and not getattr(self.config, "tune_mm_projector", False):
                self.get_mm_projector().eval()
```

it will print the warning when `tune_language_model=False` in training stage. However, I do not understand this behavior since I only want to finetune the mm_projector weights. Please give me some tips. thanks you.

**Comment by Efficient-Large-Language-Model** [comment]
First, setting `tune_language_model=False` and `tune_mm_projector=True` may have unexpected behavior. There is an issue in huggingface transformer when the later part of the model is frozen, the gradient may not propagate back correctly.

Second, it is recommended to set tune_language_model to be True. Only tuning mm_projector weights may not give you desirable results anyway.

**Comment by dragen1860** [comment]
> First, setting `tune_language_model=False` and `tune_mm_projector=True` may have unexpected behavior. There is an issue in huggingface transformer when the later part of the model is frozen, the gradient may not propagate back correctly.
> 
> Second, it is recommended to set tune_language_model to be True. Only tuning mm_projector weights may not give you desirable results anyway.

So, when finetunning on downstream tasks, it is recommeded to set `tune_language_model=True` and `tune_mmprojector=False` ?

**Comment by Efficient-Large-Language-Model** [comment]
`tune_language_model=True` and `tune_mmprojector=True`

**Comment by Lyken17** [comment]
Hi there,

Our team has moved our VILA to a new location: [NVLabs/VILA](https://github.com/NVlabs/VILA). We kindly request that you update your references to the old repository location with the new one (e.g, git clone / pipeline integration). This will ensure that you have access to the latest updates, improvements, and bug fixes for VILA project. Current repo will be archieved in one week (Jul 10th) and made private after one month (Aug 4th).

Thank you for your understanding and continued support. If your issue still exists, feel free to open a new one in the new place. We appreciate your patience and cooperation during this transition.

Best regards,
VILA Team
